src/org/sosy_lab/common/log/ConsoleLogFormatter.java

```diff
@@ -8,61 +8,32 @@
 
 package org.sosy_lab.common.log;
 
-import com.google.common.base.MoreObjects;
 import com.google.errorprone.annotations.Var;
 import java.util.Objects;
 import java.util.logging.Formatter;
-import java.util.logging.Level;
 import java.util.logging.LogRecord;
 
 /** Class to handle formatting for console output. */
-public class ConsoleLogFormatter extends Formatter {
-
-  private final boolean useColors;
+public class ConsoleLogFormatter extends AbstractColoredLogFormatter {
 
   public ConsoleLogFormatter(LoggingOptions options) {
     this(options.useColors());
   }
 
-  private ConsoleLogFormatter(@Var boolean pUseColors) {
-
-    if (pUseColors) {
-      // Using colors is only good if stderr is connected to a terminal and not
-      // redirected into a file.
-      // AFAIK there is no way to determine this from Java, but at least there
-      // is a way to determine whether stdout is connected to a terminal.
-      // We assume that most users only redirect stderr if they also redirect
-      // stdout, so this should be ok.
-      if ((System.console() == null)
-          // Windows terminal does not support colors
-          || System.getProperty("os.name", "").startsWith("Windows")
-          // https://no-color.org/
-          || System.getenv("NO_COLOR") != null) {
-        pUseColors = false;
-      }
-    }
-    useColors = pUseColors;
+  private ConsoleLogFormatter(@Var boolean useColors) {
```

Comment:
Use `@Var` only if the variable is going to me modified (it is the inverse of `final`).